blocks/api/registration.js

```diff
@@ -43,6 +43,13 @@ export function registerBlock( slug, settings ) {
 		);
 		return;
 	}
+	if ( true !== validateBlockSettings( settings ) ) {
```

Comment:
> Should we inline block settings validation for consistency with the slug validation?

Most likely I am going to make another PR and totally redo the approach, to have the validation be all in one function rather than the sort of procedural thing we have developing.

> Should we expect the validate to do something, like throw, or perhaps prefixing with is to reflect its boolean return value is more accurate, like isValidBlockSettings

Prefixing sounds good.  We can throw as well.

> If we know the function returns a boolean, do we need to be so explicit with the comparison, or can we simplify to ! isValidBlockSettings ?

I am usually a fan of the strict checks, but can definitely change it.